grass7/raster/r.mapcalc.tiled/r.mapcalc.tiled.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python
+
+############################################################################
+#
+# MODULE:	r.mapcalc.tiled
+# AUTHOR(S):	Moritz Lennert
+#
+# PURPOSE:	Run r.mapcalc over tiles of the input map
+#               to allow parallel processing
+# COPYRIGHT:	(C) 1997-2018 by the GRASS Development Team
+#
+#		This program is free software under the GNU General Public
+#		License (>=v2). Read the file COPYING that comes with GRASS
+#		for details.
+#############################################################################
+
+#%Module
+#% description: Runs r.mapcalc in parallel over tiles
+#% keyword: raster
+#% keyword: mapcalc
+#% keyword: tiling
+#%end
+#
+#%option
+#% key: expression
+#% type: string
+#% description: Expression to send to r.mapcalc
+#% required: yes
+#%end
+#
+#%option G_OPT_R_OUTPUT
+#% description: Name of raster output map resulting from expression
+#% key: output
+#% required : no
+#%end
+#
+#%option
+#% key: tile_width
+#% type: integer
+#% description: Width of tiles
+#% answer: 1000
+#% required: yes
+#%end
+#
+#%option
+#% key: tile_height
+#% type: integer
+#% description: Height of tiles
+#% answer: 1000
+#% required: yes
+#%end
+#
+#%option
+#% key: overlap
+#% type: integer
+#% description: Overlap between tiles
+#% answer: 0
+#% required: yes
+#%end
+#
+#%option
+#% key: processes
```

Comment:
```suggestion
#% key: nprocs
```
Reason: parameter compatibility with t.rast.mapcalc

Reply:
Thanks for all these corrections !

On this one, I personally prefer processes which is more in line with the general idea of spelling out parameters and avoiding abbreviations and it is the version we use in a lot of the other addon modules using parallel processing. I guess this means we should make this into a more fundamental discussion on this parameter name.

Reply:
I generally agree, but we have this much (plus testsuite) using the standard `nprocs` parameter - IMHO a task for GRASS GIS 8.0 to change to a better parameter name:

```
ag -l nprocs | grep -v testsuite
vector/v.surf.rst/main.c
lib/python/pygrass/modules/interface/module.py
lib/python/temporal/mapcalc.py
lib/python/temporal/temporal_raster3d_algebra.py
lib/python/temporal/aggregation.py
lib/python/temporal/temporal_raster_algebra.py
lib/python/temporal/extract.py
lib/python/temporal/temporal_raster_base_algebra.py
lib/python/temporal/temporal_algebra.py
lib/cdhc/doc/goodness.ps
gui/wxpython/animation/controller.py
gui/wxpython/datacatalog/tree.py
gui/wxpython/animation/provider.py
gui/wxpython/animation/dialogs.py
gui/wxpython/core/settings.py
temporal/t.rast.to.vect/t.rast.to.vect.html
temporal/t.rast.to.vect/t.rast.to.vect.py
temporal/t.rast.neighbors/t.rast.neighbors.html
temporal/t.rast.neighbors/t.rast.neighbors.py
temporal/t.rast.neighbors/test.t.rast.neighbors.sh
temporal/t.rast3d.extract/t.rast3d.extract.py
temporal/t.rast.mapcalc/t.rast.mapcalc.html
temporal/t.rast.mapcalc/test.t.rast.mapcalc.operators.sh
temporal/t.rast.mapcalc/t.rast.mapcalc.py
temporal/t.rast.mapcalc/test.t.rast.mapcalc.sh
temporal/t.rast.gapfill/t.rast.gapfill.py
temporal/t.rast3d.algebra/t.rast3d.algebra.py
temporal/t.rast.aggregate/t.rast.aggregate.py
temporal/t.rast.aggregate/t.rast.aggregate.html
temporal/t.rast.algebra/t.rast.algebra.py
temporal/t.rast.what/t.rast.what.py
temporal/t.rast.aggregate.ds/t.rast.aggregate.ds.py
temporal/t.rast.contour/t.rast.contour.py
temporal/t.rast.contour/t.rast.contour.html
temporal/t.rast3d.mapcalc/t.rast3d.mapcalc.py
temporal/t.rast3d.mapcalc/test.t.rast3d.mapcalc.sh
temporal/t.rast.extract/t.rast.extract.py
temporal/t.vect.extract/test.t.vect.extract.sh
temporal/t.vect.extract/t.vect.extract.py
raster/r.sun/main.c
raster/r.sim/r.sim.sediment/main.c
raster/r.sim/r.sim.water/main.c
raster3d/r3.gradient/main.c
```

Reply:
... of course no showstopper for me! Happy to see this new addon available

Reply:
Leaving as "processes" for now. It we decide to harmonize, I should change it also in all our other modules (i.segment.uspo, i.segment.stats, etc).